web_monitoring/diffing_server.py

```diff
@@ -325,35 +313,40 @@ def check_response_for_error(self, response):
 
 def _extract_encoding(headers, content):
     encoding = None
-    content_type = headers.get('Content-Type', '')
+    content_type = headers.get('Content-Type', '').lower()
     if 'charset=' in content_type:
         encoding = content_type.split('charset=')[-1]
-    if not encoding:
-        meta_tag_match = META_TAG_PATTERN.search(content, endpos=2048)
-        if meta_tag_match:
-            encoding = meta_tag_match.group(1).decode('ascii', errors='ignore')
-    if not encoding:
-        prolog_match = XML_PROLOG_PATTERN.search(content, endpos=2048)
-        if prolog_match:
-            encoding = prolog_match.group(1).decode('ascii', errors='ignore')
+        if encoding:
+            encoding = encoding.strip()
+    if not encoding and content:
+        # try to identify encoding using cchardet. Use up to 18kb of the
+        # content for detection. Its not necessary to use the full content
+        # as it could be huge. Also, if you use too little, detection is not
+        # accurate.
+        detected = cchardet.detect(content[:18432])
```

Comment:
Do we need to guard against possible exceptions here?

Reply:
No, it doesn't throw any exceptions from my experience.